SwiftCompilerSources/CMakeLists.txt

```diff
@@ -230,18 +230,25 @@ else()
 
   add_subdirectory(Sources)
 
-  # This is a hack to get correct dependencies from imported C++ headers:
-  # step 1: generate a dummy source file, which just includes all headers defined in include/swift/module.modulemap
-  add_custom_command(
-    OUTPUT "${CMAKE_CURRENT_BINARY_DIR}/HeaderDependencies.cpp"
-    COMMAND
-        "sed"
-        -n -e "/header/!d" -e "s/.*header/#include/" -e "w ${CMAKE_CURRENT_BINARY_DIR}/HeaderDependencies.cpp"
-        "${CMAKE_CURRENT_SOURCE_DIR}/../include/swift/module.modulemap"
-    DEPENDS "${CMAKE_CURRENT_SOURCE_DIR}/../include/swift/module.modulemap"
-    COMMENT "Generate HeaderDependencies.cpp"
-)
-  
+  # TODO: generate this dynamically through the modulemap; this cannot use `sed`
+  # as that is not available on all paltforms (e.g. Windows).
```

Comment:
you can keep the "step 1:" part of the comment.
Otherwise there's just a hanging "step 2:" down below and someone will be searching for step one.

Reply:
That seems reasonable.

Reply:
- [x]